### PR TITLE
Turning off logger at the class-constructor level

### LINKED ISSUE #316: feat: Allow logger to be disabled at the class-constructor level
**Is your feature or enhancement request related to a problem or limitation? Please describe**

- After #310 was implemented...
- It would be great if we allowed the logger to be disabled  as part of the class constructor

**Describe your enhancement idea**

As an example, someone could do `Datasets(profile, log=False)`

**Describe alternatives you've considered**

```python
profile = ProfileManager().load(profile_type="zosmf", validate_schema=False)
ds = Datasets(profile)
Log.close_all()
console = Console(profile)
jobs = Jobs(profile)
```

The above will run off the logger for Datasets abut keep it open for the other classes since they are initialized after the `Log.close_all()` call

**Provide any additional context**

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
**What It Does**

Resolves #316 

**How to Test**

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/core/zowe/core_for_zowe_sdk/sdk_api.py [hunk 1]
```diff
@@ -33,12 +33,15 @@ class SdkApi:
         Name of the logger (same as the filename by default)
     """
 
-    def __init__(self, profile: dict, default_url: str, logger_name: str = __name__):
+    def __init__(self, profile: dict, default_url: str, logger_name: str = __name__, log = False):
```
Comment (t1m0thyj) [inline src/core/zowe/core_for_zowe_sdk/sdk_api.py:36]: I think we want logging enabled by default, so the default value of the `log` argument should be True.

File: src/zos_console/zowe/zos_console_for_zowe_sdk/console.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ class Console(SdkApi):
        A profile in dict (json) format
     """
 
-    def __init__(self, connection: dict):
+    def __init__(self, connection: dict, log = False):
         super().__init__(connection, "/zosmf/restconsoles/consoles/defcn", logger_name=__name__)
```
Comment (t1m0thyj) [inline src/zos_console/zowe/zos_console_for_zowe_sdk/console.py:31]: We need to pass the `log` argument on to the SdkApi class. This comment applies to all the other SDKs too 😋 
```suggestion
        super().__init__(connection, "/zosmf/restconsoles/consoles/defcn", logger_name=__name__, log=log)
```

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Thanks @aadityasinha-dotcom! Left a few comments

File: src/zos_console/zowe/zos_console_for_zowe_sdk/console.py [hunk 3]
```diff
@@ -27,8 +27,8 @@ class Console(SdkApi):
        A profile in dict (json) format
     """
 
-    def __init__(self, connection: dict):
-        super().__init__(connection, "/zosmf/restconsoles/consoles/defcn", logger_name=__name__)
+    def __init__(self, connection: dict, log = False):
```
Comment (traeok) [inline src/zos_console/zowe/zos_console_for_zowe_sdk/console.py:30]: Can the default value for this argument be updated to `True`?
```suggestion
    def __init__(self, connection: dict, log = True):
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ class USSFiles(SdkApi):
         The z/OSMF connection object (generated by the ZoweSDK object)
     """
 
-    def __init__(self, connection: dict):
+    def __init__(self, connection: dict, log = False):
```
Comment (traeok) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:37]: Same comment applies to the `USSFiles` class

**Review by traeok — CHANGES_REQUESTED** [review]

**Review by traeok — APPROVED** [review]
LGTM, thanks @aadityasinha-dotcom for the enhancement!

**Review by t1m0thyj — DISMISSED** [review]

File: src/core/zowe/core_for_zowe_sdk/sdk_api.py [hunk 5]
```diff
@@ -33,12 +33,15 @@ class SdkApi:
...
 
-    def __init__(self, profile: dict, default_url: str, logger_name: str = __name__):
+    def __init__(self, profile: dict, default_url: str, logger_name: str = __name__, log = True):
         session = Session(profile)
         self.session: ISession = session.load()
 
         self.logger = Log.register_logger(logger_name)
 
+        if log == False:
+            Log.close(self.logger)
```
[the remark this answers is not in the document]
Comment (aadityasinha-dotcom) [reply src/core/zowe/core_for_zowe_sdk/sdk_api.py:45]: Is it required to add unit tests for other classes or only for sdk_api?

[the remark this answers is not in the document]
Comment (t1m0thyj) [reply src/core/zowe/core_for_zowe_sdk/sdk_api.py:45]: I think just for SdkApi is fine 😋

File: tests/unit/core/test_sdk_api.py [hunk 6]
```diff
@@ -25,6 +25,11 @@ def test_object_should_be_instance_of_class(self):
         sdk_api = SdkApi(self.basic_props, self.default_url)
         self.assertIsInstance(sdk_api, SdkApi)
 
+    def test_object_should_be_instance_with_logger_set_to_false(self):
```
Comment (aadityasinha-dotcom) [inline tests/unit/core/test_sdk_api.py:28]: Need some suggestion on the unit test :relaxed: 
When I was testing with `log=False` the logger is not returning `None` (I guess maybe it should not)

Comment (traeok) [reply tests/unit/core/test_sdk_api.py:28]: Hi @aadityasinha-dotcom,

I believe the logger is still present, but the Logger will have its `disabled` class variable set to `true` if logging is turned off:
https://github.com/zowe/zowe-client-python-sdk/blob/eb1ad71abc2de293d42e8bd1c34cf2fb42b80abb/src/core/zowe/core_for_zowe_sdk/logger.py#L90-L100

Comment (aadityasinha-dotcom) [reply tests/unit/core/test_sdk_api.py:28]: I see, thank you for the help :+1:

Comment (aadityasinha-dotcom) [inline tests/unit/core/test_sdk_api.py:28]: Did changes to the test now it is verifying logger.disabled to True :relaxed:

**Comment by aadityasinha-dotcom** [comment]
Hello, I have resolved the lint errors also, please review it :relaxed:

**Review by traeok — APPROVED** [review]
LGTM, thanks @aadityasinha-dotcom!

**Review by zFernand0 — APPROVED** [review]
Thanks for the implementation @aadityasinha-dotcom 

I think it'd be good to have a quick Changelog entry about this. 🥳 

```markdown
## Recent Changes

### Enhancements

- Turning of ......... [#316] (https://github.com/zowe/zowe-client-python-sdk/issues/316)

```
LGTM! 😋

File: CHANGELOG.md [hunk 7]
```diff
@@ -8,6 +8,10 @@ All notable changes to the Zowe Client Python SDK will be documented in this fil
 
 - Fixed Core SDK package referencing a non-existent version of Secrets SDK.
 
+### Enhancements
```
Comment (aadityasinha-dotcom) [inline CHANGELOG.md:11]: Added changelog also :smile:

Comment (zFernand0) [reply CHANGELOG.md:11]: Sorry for not being explicit, but the changelog has to be towards the top of the file.
and you must include the header `## Recent Changes` 😓 

my apologies for that 😢

**Pull request merged**